Lib/test/_test_multiprocessing.py

```diff
@@ -4119,29 +4146,42 @@ def test_shared_memory_ShareableList_basics(self):
             empty_sl.shm.unlink()
 
     def test_shared_memory_ShareableList_pickling(self):
+        for proto in range(pickle.HIGHEST_PROTOCOL + 1):
+            with self.subTest(proto=proto):
+                sl = shared_memory.ShareableList(range(10))
+                self.addCleanup(sl.shm.unlink)
+
+                serialized_sl = pickle.dumps(sl, protocol=proto)
+                deserialized_sl = pickle.loads(serialized_sl)
+                self.assertIsInstance(
+                    deserialized_sl, shared_memory.ShareableList)
+                self.assertEqual(deserialized_sl[-1], 9)
+                self.assertIsNot(sl, deserialized_sl)
+                deserialized_sl[4] = "changed"
+                self.assertEqual(sl[4], "changed")
```

Comment:
It would be nice to test also the other direction as in the doctest: change `sl` and read `deserialized_sl`.